[CI] CrossClusterSearchUnavailableClusterIT testSearchSkipUnavailable failing

**Build Scans:**
- [elasticsearch-periodic #6200 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/rgzlgzti6wfgc)
- [elasticsearch-periodic #6172 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/yesyap5kuozio)
- [elasticsearch-periodic #6151 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/vyk2clz4b4sqi)
- [elasticsearch-periodic #6130 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/4hag7ytif5kqa)

**Reproduction Line:**
```
./gradlew ":qa:ccs-unavailable-clusters:javaRestTest" --tests "org.elasticsearch.search.CrossClusterSearchUnavailableClusterIT.testSearchSkipUnavailable" -Dtests.seed=38CE078987E78562 -Dtests.locale=en-BI -Dtests.timezone=Etc/GMT+7 -Druntime.java=17 -Dtests.fips.enabled=true
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.CrossClusterSearchUnavailableClusterIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSearchSkipUnavailable),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
#1:
	in [elasticsearch[node0][transport_worker][T#13]][testSearchSkipUnavailable]
	org.elasticsearch.transport.OutboundHandler.lambda$sendResponse$1(OutboundHandler.java:158)
	org.elasticsearch.core.IOUtils.close(IOUtils.java:71)
	org.elasticsearch.core.IOUtils.close(IOUtils.java:87)
	org.elasticsearch.core.IOUtils.close(IOUtils.java:63)
	org.elasticsearch.core.Releasables.close(Releasables.java:73)
	org.elasticsearch.core.Releasables.close(Releasables.java:41)
	org.elasticsearch.core.Releasables$3.close(Releasables.java:129)
	org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:217)
	org.elasticsearch.transport.OutboundHandler$1.onResponse(OutboundHandler.java:257)
	org.elasticsearch.transport.OutboundHandler$1.onResponse(OutboundHandler.java:253)
	org.elasticsearch.transport.netty4.Netty4Utils.lambda$addLis
[truncated]
```

**Issue Reasons:**
- [8.x] 4 failures in test testSearchSkipUnavailable (0.9% fail rate in 434 executions)
- [8.x] 4 failures in step openjdk17_checkpart1_java-fips-matrix (40.0% fail rate in 10 executions)
- [8.x] 4 failures in pipeline elasticsearch-periodic (36.4% fail rate in 11 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/qa/ccs-unavailable-clusters/src/javaRestTest/java/org/elasticsearch/search/CrossClusterSearchUnavailableClusterIT.java
+++ b/qa/ccs-unavailable-clusters/src/javaRestTest/java/org/elasticsearch/search/CrossClusterSearchUnavailableClusterIT.java
@@ -101,8 +101,8 @@ public class CrossClusterSearchUnavailableClusterIT extends ESRestTestCase {
                 TransportSearchAction.TYPE.name(),
                 EsExecutors.DIRECT_EXECUTOR_SERVICE,
                 SearchRequest::new,
-                (request, channel, task) -> channel.sendResponse(
-                    new SearchResponse(
+                (request, channel, task) -> {
+                    var searchResponse = new SearchResponse(
                         SearchHits.empty(new TotalHits(0, TotalHits.Relation.EQUAL_TO), Float.NaN),
                         InternalAggregations.EMPTY,
                         null,
@@ -117,8 +117,13 @@ public class CrossClusterSearchUnavailableClusterIT extends ESRestTestCase {
                         100,
                         ShardSearchFailure.EMPTY_ARRAY,
                         SearchResponse.Clusters.EMPTY
-                    )
-                )
+                    );
+                    try {
+                        channel.sendResponse(searchResponse);
+                    } finally {
+                        searchResponse.decRef();
+                    }
+                }
             );
             newService.registerRequestHandler(
                 ClusterStateAction.NAME,

